### PR TITLE
Fix layouts transient state doc and tweak formatting of hint 

### PR DESCRIPTION
**Pull request by justbur**

### CODE REVIEW

**Comment by bmag** [comment]
Thanks, I've tried this locally and there's an issue I want to point out. Toggling `?` hides also the available layouts (the first line beneath the "Layouts Transient State" header), but they are supposed to be displayed even when the rest of the help text (the available keys) is hidden.

**Comment by justbur** [comment]
@bmag it always shows them for me. Is there any more information you could provide?

**Comment by bmag** [comment]
Disabling my layers and additional packages revealed that the problem is not because of spacemacs. The culprit is `tabbar-ruler`, which I installed as an additional package.

**Comment by justbur** [comment]
@bmag Thanks

**Comment by synic** [comment]
There's a request to have a way to disable the "Color Guide" from transient states.  It makes sense... it'll be helpful the first time, but never again after that.

**Comment by justbur** [comment]
@synic we were thinking about removing it altogether. I agree that it clutters things up. But it is useful for new people...

**Comment by nixmaniack** [comment]
I miss the default spacemacs's coloring of transient state's titles. Can we add this for the time being till we figure out proper solution?

```diff
@@ -72,7 +72,8 @@
                (caption (concat (number-to-string (if (eq 9 pos) 0 (1+ pos)))
                                 ":" string-name)))
           (if current
-              (concat (when current "[") caption (when current "]"))
+              (propertize  (concat (when current "[") caption (when current "]"))
+                           'face 'spacemacs-micro-state-binding-face)
             caption)))
 
       (defvar spacemacs--layouts-ms-documentation
@@ -162,7 +163,7 @@
                   (setq spacemacs/layouts-transient-state/hint
                         `(concat
                           ,(propertize "Layouts Transient State\n"
-                                       'face 'spacemacs-transient-state-title-face)
+                                       'face 'spacemacs-micro-state-header-face)
                           (spacemacs//layouts-ms-doc))))
                 t)
 
```

**Comment by justbur** [comment]
@synic Added the option

@nixmaniack I put in the code to match the old formatting but commented it out because I don't think it looks good with the hydra colors. The title inherits from the mode-line face now, which will at least make it not clash with the theme.

File: core/core-dotspacemacs.el [hunk 1]
```diff
@@ -218,6 +218,9 @@ can be toggled through `toggle-transparency'.")
 transparency level of a frame when it's inactive or deselected. Transparency
 can be toggled through `toggle-transparency'.")
 
+(defvar dotspacemacs-show-transient-state-color-guide t
+  "If non nil unicode symbols are displayed in the mode line. (default t)")
```
Comment (nixmaniack) [inline core/core-dotspacemacs.el]: docstring is wrong.

Comment (justbur) [inline core/core-dotspacemacs.el]: thanks

**Comment by justbur** [comment]
@syl20bnr I'm done with this unless there are more comments.

**Comment by syl20bnr** [comment]
Great improvements :purple_heart: 
Thank you :100: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**